### PR TITLE
E2E GitHub actions

### PR DESCRIPTION
**Pull request by elliothursh**
## Describe your changes
Adds Github Actions for e2e tests

## Issue link
N/A

## Checklist before requesting a review

- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: E2E Tests
+on:
+  push:
+jobs:
+  e2e:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        e2e-config:
+          - next_js
```
Comment (psugihara) [inline .github/workflows/e2e.yml:10]: so cool

**Review by psugihara — COMMENTED** [review]
beautiful but looks like the runs don't terminate for some reason

File: scripts/e2e_test/e2e_test.rb [hunk 2]
```diff
@@ -21,7 +21,7 @@ class TestRunner
 
   NUM_RUNS_TO_KEEP = 5
   PROJECT_ROOT = File.expand_path("#{__dir__}/../..")
-  TEST_ROOT = File.expand_path("#{PROJECT_ROOT}/../mailing_e2e_tests")
+  TEST_ROOT = File.expand_path("#{PROJECT_ROOT}/tmp/e2e_test")
```
Comment (elliothursh) [inline scripts/e2e_test/e2e_test.rb:24]: moved this to a local tmp dir just to not create a random folder outside of this project.

File: scripts/e2e_test/e2e_test.rb [hunk 3]
```diff
@@ -161,7 +161,7 @@ def run_jest_tests
       FileUtils.cp_r(JEST_TESTS_DIR, './jest_tests')
 
       # run yarn jest --rootDir=jest_tests
-      system('yarn jest --rootDir=jest_tests -c jest_tests/jest.config.js')
+      system('yarn jest --rootDir=jest_tests -c jest_tests/jest.config.js --detectOpenHandles --forceExit')
```
Comment (elliothursh) [inline scripts/e2e_test/e2e_test.rb:164]: testing out using forceExit as a stop gap.

File: scripts/e2e_test/e2e_test.rb (hunk 2 above)
Comment (psugihara) [reply scripts/e2e_test/e2e_test.rb:24]: it does that so that it doesn't accidentally use anything from the project's node_modules

Comment (alexfarrill) [reply scripts/e2e_test/e2e_test.rb:24]: +1 you will want to move this outside of the project directory

Comment (alexfarrill) [reply scripts/e2e_test/e2e_test.rb:24]: does github not let you do that?

Comment (psugihara) [reply scripts/e2e_test/e2e_test.rb:24]: the other option would be nuking "#{PROJECT_ROOT}/node_modules" but that would probably have other reprecussions

File: .github/workflows/e2e.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  push:
+jobs:
+  e2e:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        e2e-config:
+          - standalone
+          # - turbo
```
Comment (alexfarrill) [inline .github/workflows/e2e.yml:12]: why doesn't turbo work?

**Review by alexfarrill — APPROVED** [review]

**Review by psugihara — APPROVED** [review]

File: .github/workflows/e2e.yml (hunk 4 above)
Comment (elliothursh) [reply .github/workflows/e2e.yml:12]: there's a type issue with react i believe

Comment (alexfarrill) [reply .github/workflows/e2e.yml:12]: dang

Comment (alexfarrill) [reply .github/workflows/e2e.yml:12]: that's cool that you can configure it separately from what you run locally

**Pull request merged**